bridges/AnidexBridge.php

```diff
@@ -0,0 +1,207 @@
+					'Music - Video' => '11',
+					'Games' => '12',
+					'Applications' => '13',
+					'Pictures' => '14',
+					'Adult Video' => '15',
+					'Other' => '16'
+				)
+			),
+			'lang_id' => array(
+				'name' => 'Language',
+				'type' => 'list',
+				'values' => array(
+					'All languages' => '0',
+					'English' => '1',
+					'Japanese' => '2',
+					'Polish' => '3',
+					'Serbo-Croatian' => '4',
+					'Dutch' => '5',
+					'Italian' => '6',
+					'Russian' => '7',
+					'German' => '8',
+					'Hungarian' => '9',
+					'French' => '10',
+					'Finnish' => '11',
+					'Vietnamese' => '12',
+					'Greek' => '13',
+					'Bulgarian' => '14',
+					'Spanish (Spain)' => '15',
+					'Portuguese (Brazil)' => '16',
+					'Portuguese (Portugal)' => '17',
+					'Swedish' => '18',
+					'Arabic' => '19',
+					'Danish' => '20',
+					'Chinese (Simplified)' => '21',
+					'Bengali' => '22',
+					'Romanian' => '23',
+					'Czech' => '24',
+					'Mongolian' => '25',
+					'Turkish' => '26',
+					'Indonesian' => '27',
+					'Korean' => '28',
+					'Spanish (LATAM)' => '29',
+					'Persian' => '30',
+					'Malaysian' => '31'
+				)
+			),
+			'group_id' => array(
+				'name' => 'Group ID',
+				'type' => 'number'
+			),
+			'r' => array(
+				'name' => 'Hide Remakes',
+				'type' => 'checkbox'
+			),
+			'b' => array(
+				'name' => 'Only Batches',
+				'type' => 'checkbox'
+			),
+			'a' => array(
+				'name' => 'Only Authorized',
+				'type' => 'checkbox'
+			),
+			'q' => array(
+				'name' => 'Keyword',
+				'description' => 'Keyword(s)',
+				'type' => 'text'
+			),
+			'h' => array(
+				'name' => 'Adult content',
+				'type' => 'list',
+				'values' => array(
+					'No filter' => '0',
+					'Hide +18' => '1',
+					'Only +18' => '2'
+				)
+			)
+		)
+	);
+
+	public function collectData() {
+
+		// Build Search URL from user-provided parameters
+		$search_url = self::URI . '?s=upload_timestamp&o=desc';
+		foreach (array('id', 'lang_id', 'group_id') as $param_name) {
+			$param = $this->getInput($param_name);
+			if (!empty($param) && intval($param) != 0 && ctype_digit(str_replace(',', '', $param))) {
+				$search_url .= '&' . $param_name . '=' . $param;
+			}
+		}
+		foreach (array('r', 'b', 'a') as $param_name) {
+			$param = $this->getInput($param_name);
+			if (!empty($param) && boolval($param)) {
+				$search_url .= '&' . $param_name . '=1';
+			}
+		}
+		$query = $this->getInput('q');
+		if (!empty($query)) {
+			$search_url .= '&q=' . urlencode($query);
+		}
+		$opt = array();
+		$h = $this->getInput('h');
+		if (!empty($h) && intval($h) != 0 && ctype_digit($h)) {
+			$opt[CURLOPT_COOKIE] = 'anidex_h_toggle=' . $h;
+		}
+
+		// Retrieve torrent listing from search results, which does not contain torrent description
+		$html = getSimpleHTMLDOM($search_url, array(), $opt)
+		or returnServerError('Could not request Anidex: '.$search_url);
+		$links = $html->find('a');
+		$results = array();
+		foreach ($links as $link)
```

Comment:
Why not merge the two loops ?
Moreover, returning an error in case of no element being found is probably not the right way to go: If I am searching for something that I know will appear in the future but does not yet (like a new release), the bridge will fail

Reply:
Seems legit, I wrote this with the idea of failing on purpose if HTML changes in result page, because no results e.g. for a category is not supposed to happen. I'll change it to:
```php
if (empty($results) && empty($this->getInput('q')))
	returnServerError('No results from Anidex: '.$search_url);
```
This way, it will not fail when a keyword is specified.

The loops are not merged because I think it makes the bridge easier to read and maintain. First it eliminates duplicates, then it checks for empty results, and finally it process the results. Is it OK to keep it this way?